server/backend/rpcserver.go

```diff
@@ -69,6 +76,46 @@ func (s *CampusServer) GetNewsSources(ctx context.Context, _ *emptypb.Empty) (ne
 	return &pb.NewsSourceArray{Sources: resp}, nil
 }
 
+// SearchRooms returns all rooms that match the given search query.
+func (s *CampusServer) SearchRooms(ctx context.Context, req *pb.SearchRoomsRequest) (*pb.SearchRoomsReply, error) {
+	if err := s.checkDevice(ctx); err != nil {
+		return nil, err
+	}
+	var res []struct { // struct to scan database query into
+		model.RoomfinderRooms
+		Campus string
+		Name   string
+	}
+	err := s.db.Raw("SELECT r.*, a.campus, a.name "+
+		"FROM roomfinder_rooms r "+
+		"LEFT JOIN roomfinder_building2area a ON a.building_nr = r.building_nr "+
+		"WHERE MATCH(room_code, info, address) AGAINST(?)", req.Query).Scan(&res).Error
+	if errors.Is(err, gorm.ErrRecordNotFound) {
+		return nil, status.Error(codes.NotFound, err.Error())
```

Comment:
IDK, I don't think this is per se an error, but the result set should just be empty.